lnwallet/wallet.go

```diff
@@ -2525,12 +2536,20 @@ func (l *LightningWallet) CancelRebroadcast(txid chainhash.Hash) {
 // CoinSource is a wrapper around the wallet that implements the
 // chanfunding.CoinSource interface.
 type CoinSource struct {
-	wallet *LightningWallet
+	wallet    *LightningWallet
+	allowUtxo AllowUtxoFn
 }
 
+// AllowUtxoFn is a function type which adds restrictions to the selection
+// of the utxos used as a coin source.
+type AllowUtxoFn func(Utxo) bool
```

Comment:
I don't think we need this typedef. Better to just leave the function definition in line.